core/src/test/scala/unit/kafka/log/LogSegmentTest.scala

```diff
@@ -181,24 +180,28 @@ class LogSegmentTest {
     assertFalse(reopened.timeIndex.isFull)
     assertFalse(reopened.offsetIndex.isFull)
 
-    assertFalse(reopened.shouldRoll(messagesSize = 1024,
-      maxTimestampInMessages = RecordBatch.NO_TIMESTAMP,
-      maxOffsetInMessages = 100L,
-      now = time.milliseconds()))
+    val logConfig = LogConfig(Map(
+      LogConfig.SegmentMsProp -> maxSegmentMs,
+      LogConfig.SegmentBytesProp -> Int.MaxValue
+    ).asJava)
+
+    var rollParams = RollParams(logConfig, getLogAppendInfo(100L), 1024, time.milliseconds())
```

Comment:
Why don't we use the `RollParams` that takes the values explicitly instead of creating `LogConfig` and `AppendInfo` in this case? It doesn't seem like it helps in this case.

Reply:
@junrao @ijuma  Thanks for the reviews. Addressed review comment and rebased the PR.